### DISCUSSION TITLE
how to make OptionList scrollable if i have so many items on it?

### DISCUSSION THREAD

**Discussion by MohamedElgamal** [Textualize/textual#3367] [Q&A] [answered]
Hello everyone, 
i added hundreds of items to OptionList to select one from them but it is not scrollable only the visible items is appear to user and others not.

**Comment by davep** [comment, marked as answer]
This is another good example of a question that deserves [a minimal runnable example of the problem](https://github.com/Textualize/textual/discussions/3361). For example, sans some other styling, the `OptionList` likely is `height: auto`. So this example would show the problem:

```python
from textual.app import App, ComposeResult
from textual.widgets import OptionList

class OLScrollApp(App[None]):

    def compose(self) -> ComposeResult:
        yield OptionList(*[f"This is option number {n}" for n in range(5000)])

if __name__ == "__main__":
    OLScrollApp().run()
```

but with styling to tell `OptionList` how tall it should be, scrolling will work as expected:

```python
from textual.app import App, ComposeResult
from textual.widgets import OptionList

class OLScrollApp(App[None]):

    CSS = """
    OptionList {
        height: 1fr;
    }
    """

    def compose(self) -> ComposeResult:
        yield OptionList(*[f"This is option number {n}" for n in range(5000)])

if __name__ == "__main__":
    OLScrollApp().run()
```
